pkg/controller/gather_job.go

```diff
@@ -39,10 +42,28 @@ func (d *GatherJob) Gather(ctx context.Context, kubeConfig, protoKubeConfig *res
 		return err
 	}
 
+	configClient, err := configv1client.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+
 	gatherProtoKubeConfig, gatherKubeConfig, metricsGatherKubeConfig, alertsGatherKubeConfig := prepareGatherConfigs(
 		protoKubeConfig, kubeConfig, d.Impersonate,
 	)
 
+	tpEnabled, err := isTechPreviewEnabled(ctx, configClient)
+	if err != nil {
+		klog.Error("can't read cluster feature gates: %v", err)
+	}
+	var gatherConfig v1alpha1.GatherConfig
+	if tpEnabled {
+		insightsDataGather, err := configClient.ConfigV1alpha1().InsightsDataGathers().Get(ctx, "cluster", metav1.GetOptions{}) //nolint: govet
```

Comment:
Wrt `//nolint: govet` - I am failing to see why the linting tells me that the `err` shadows the one at line 40.